webservice/tests/test_oauth2.py

```diff
@@ -213,3 +216,33 @@ def test_fetch_token_from_auth(self):
         adapter = self.webservice._get_adapter()
         token = adapter._fetch_token_from_authorization(code)
         self.assertEqual("cool_token", token["access_token"])
+
+    def test_oauth2_flow_compute(self):
+        # Test with current configuration
+        env = os.environ
+        self.assertIn("auth_type = oauth2\n", env["SERVER_ENV_CONFIG"])
+        self.assertIn("oauth2_flow = web_application\n", env["SERVER_ENV_CONFIG"])
+        server_env_mixin.serv_config = server_env._load_config()  # Reload env vars
+        self.webservice.invalidate_recordset()  # Force recomputation when reading
+        self.assertEqual(self.webservice.oauth2_flow, "web_application")
+
+        # Update configuration: ``auth_type`` is changed to ``none``
+        env["SERVER_ENV_CONFIG"] = env["SERVER_ENV_CONFIG"].replace(
+            "auth_type = oauth2", "auth_type = none"
+        )
+        server_env_mixin.serv_config = server_env._load_config()  # Reload env vars
+        self.webservice.invalidate_recordset()  # Clear cache => read forces compute
```

Comment:
No, we don't lose the update on the form: it is lost in tests because Odoo will fetch the cached value, but that doesn't happen when reading the value in UI.